src/test/java/com/skalashynski/spring/springboot/service/StudentServiceTest.java

```diff
@@ -0,0 +1,177 @@
+package com.skalashynski.spring.springboot.service;
+
+import com.skalashynski.spring.springboot.entity.Student;
+import com.skalashynski.spring.springboot.exception.ApiException;
+import com.skalashynski.spring.springboot.repository.StudentRepository;
+import com.skalashynski.spring.springboot.service.impl.StudentServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.BDDMockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class StudentServiceTest {
+    private StudentService studentService;
+    @Mock
+    private StudentRepository studentRepository;
+
+    private Student student;
+
+    @BeforeEach
+    void init() {
+        studentService = new StudentServiceImpl(studentRepository);
+        student = new Student(1L, "firstName", "lastName",
+                LocalDate.of(2000, 1, 1), LocalDateTime.now());
+    }
+
+    @Test
+    void saveTest_whenMethodInvokes_thenStudentEntitySavesIntoRepAndReturns() {
```

Comment:
You can use `@DisplayName` annotation and provide descriptive test name
Apply please for all tests.

Reply:
done